src/class/object_genome_variants.php

```diff
@@ -245,9 +245,19 @@ function checkFields ($aData, $zData = false)
 
         if ($_AUTH['level'] >= LEVEL_CURATOR) {
             $this->aCheckMandatory[] = 'effect_concluded';
-        } elseif (isset($aData['effect_reported']) && $aData['effect_reported'] === '0') {
-            // Submitters must fill in the variant effect field; '0' is not allowed for them.
-            unset($aData['effect_reported']);
+        }
+
+        if (isset($aData['effect_reported']) && $aData['effect_reported'] === '0') {
+            // `effect_reported` is not allowed to be '0' (Not classified) when user is a submitter
+            // or when the variant has status '9' (Public).
+            if ($_AUTH['level'] < LEVEL_CURATOR) {
+                // Remove the mandatory `effect_reported` field to throw an error.
+                unset($aData['effect_reported']);
+            } elseif (isset($aData['statusid']) && intval($aData['statusid']) == 9) {
+                // Show error for curator/manager trying to publish variant without effect.
+                lovd_errorAdd('effect_reported', 'The \'Affects function (reported)\' field ' .
+                    'must be filled when variant status is Public');
+            }
```

Comment:
- To prevent problems later, please use `STATUS_PUBLIC` instead of `9`.
  We could decide on applying this method as well to the variant effects?
- We don't need (or use) `intval()` anywhere. Would there be an advantage? We always rely on the typecasting of the `==` operator.
- I think the error message could be improved, as it's pretty much the same as the user will get when the field is really not filled in. Perhaps add that `Not classified` is not allowed for public variants?